go-controller/pkg/ovn/obj_retry.go

```diff
@@ -93,73 +247,53 @@ func NewRetryObjs(
 	}
 }
 
-// setRetryObjWithNoBackoff sets an object's backoff to be retried
-// immediately during the next retry iteration
-// Used only for testing right now
-func (r *retryObjs) setRetryObjWithNoBackoff(key string) {
-	r.retryMutex.Lock()
-	defer r.retryMutex.Unlock()
-	if entry, ok := r.entries[key]; ok {
-		entry.backoffSec = noBackoff
-		return
-	}
-
-	klog.Errorf("Unable to find entry with key %s", key)
-}
-
-// addRetryObjWithAddNoBackoff adds an object to be retried immediately for add
-func (r *retryObjs) addRetryObjWithAddNoBackoff(obj interface{}) {
-	key, err := getResourceKey(r.oType, obj)
-	if err != nil {
-		klog.Errorf("Could not get the key of %s %v: %v", r.oType, obj, err)
-		return
-	}
-	r.initRetryObjWithAddBackoff(obj, key, noBackoff)
-	r.unSkipRetryObj(key)
+func (r *retryObjs) lockKey(key string) LockedKey {
+	return r.retryEntries.LockKey(key)
 }
 
-// initRetryObjWithAdd creates a retry entry for an object that is being added,
-// so that, if it fails, the add can be potentially retried later.
-// initially it is marked as skipped for the retry loop (ignore = true).
-func (r *retryObjs) initRetryObjWithAdd(obj interface{}, key string) {
-	r.initRetryObjWithAddBackoff(obj, key, initialBackoff)
+func (r *retryObjs) unlockKey(key LockedKey) {
+	r.retryEntries.UnlockKey(key)
```

Comment:
I think we should avoid leaking the concept of a locked key outside of retryObjs, at least on this form, but this is a bit of a mess, there is no clear interface of what is meant to be private to retryObjs (or testing only) or public to the outside world, and this is because we are not moving this to it's own package. 

Also unclear on what is the interaction between Controller and retry_objs, they seem to be tightly coupled, but at least it is hopefuly confined within this file.

So my suggestion is, for now:

- Capitalize methods meant to be used outside of retryObjs and do no leak the concept of locked key there:
  * AddRetryObj* 
  * DeleteRetryObj*
  * ...
-  keep lower case for other private methods meant to be used within retryObjs (or controller methods within this file) that need a locked key
   * initRetryObj*
   * ...
- move testing methods that you have in pod_test.go but you use everywhere to obj_retry_test.go 

I would remove these lock/unlock methods once it's clear they are only need privately, within obj_retry we can just use r.retryEntries.LockKey, at least until this is moved to it's own package.